lib/gooddata/core/gd_logger.rb

```diff
@@ -0,0 +1,56 @@
+# encoding: UTF-8
+#
+# Copyright (c) 2010-2017 GoodData Corporation. All rights reserved.
+# This source code is licensed under the BSD-style license found in the
+# LICENSE file in the root directory of this source tree.
+
+require 'logger'
+
+module GoodData
+  class GDLogger < Logger
+    attr_accessor :log_to_splunk
+    attr_accessor :splunk_level
+
+    def initialize(*args)
+      super(*args)
+      @log_to_splunk = false
+      @splunk_level = 1
+    end
+
+    def token_to_severity(severity)
+      if severity.is_a?(Integer)
+        @level = severity
+      else
+        case severity
+        when :debug
+          @level = DEBUG
+        when :info
+          @level = INFO
+        when :warn
+          @level = WARN
+        when :error
+          @level = ERROR
+        when :fatal
+          @level = FATAL
+        when :unknown
+          @level = UNKNOWN
+        else
+          raise ArgumentError, "invalid log level: #{severity}"
+        end
+      end
+    end
+
+    [:unknown, :debug, :fatal, :error, :info, :warn].each do |severity|
+      define_method severity do |*args, &block|
+        add(token_to_severity(severity), nil, nil, &block) if args.empty?
+        add(token_to_severity(severity), nil, args[0], &block) if args.length == 1
+        add(token_to_severity(severity), nil, args[0], args[1], &block) if args.length == 2
+      end
+    end
+
+    def add(severity, message, progname, splunk_severity=10, &block)
```

Comment:
Style/SpaceAroundEqualsInParameterDefault: Surrounding space missing in default value assignment.